### ISSUE TITLE
docs: readthedocs site renders building.rst wrong

### ISSUE THREAD

**Issue by mythi** [gramineproject/gramine#1783]
### Description of the problem

If I look at `Build Gramine`, I can see lots of nested `Note`s. Maybe other pages do not look correct either but I'm not able check all of them.

### Steps to reproduce

Check https://gramine.readthedocs.io/en/latest/devel/building.html. For example, the steps where `ninja` is called is not "top level" anymore.

### Expected results

The text renders correctly.

### Gramine commit hash

HEAD

**Comment by dimakuv** [comment]
Wow. I don't see any issue/any changes in our sources, e.g.:
```
...

.. note::

   If you plan to contribute changes to Gramine, then you should always build it
   with ``--werror`` added to the invocation above.

.. note::

   If you invoked ``meson setup`` once, the next invocation of this command will
   *not* have any effect. Instead, to change the build configuration, use
   ``meson configure``. For example, if you built with ``meson setup build/
   -Dsgx=disabled`` first and now want to enable SGX, type ``meson configure
   build/ -Dsgx=enabled``.

Set ``-Ddirect=`` and ``-Dsgx=`` options to ``enabled`` or ``disabled``
according to whether you built the corresponding PAL (the snippet assumes you
built both).

...
```

So this is apparently some misunderstanding of the RestructedText `note` directive on our side. Or a bug in the RestructuredText renderer that we updated recently: https://github.com/gramineproject/gramine/commit/97448b451829f12b339583d8f68d8be74fbb4f1f

@woju Do you have any ideas?

**Comment by woju** [comment]
This is almost certainly this thing: https://github.com/sphinx-doc/sphinx/issues/10121 (and a duplicate: https://github.com/sphinx-doc/sphinx/issues/9923).
It's a regression between Sphinx and Docutils, and it was triggered by updating to docutils>=0.18 without updating to Sphinx >= 4.5. (Sphinx was fixed in 4.5, but we have only 3.4 for Debian `oldstable` compatibility).

One solution would be to downgrade docutils to `<0.18`, as it were.

Another solution would be to redefine how we support different Sphinx versions with the aim of upgrading RTD version of sphinx. @dimakuv @mkow can we have a policy that:
- For packaging we support old version of sphinx (currently 3.4 for `oldstable`) to build `man` only.
- We support `html` building with a version from Debian `stable` (currently 5.3) for people to be able to build html locally for development and review. We then reconfigure RTD to use this newer version.

This will result in such a situation:

- Debian `stable`: can build `html` and `man`
- Debian `oldstable`, all Ubuntu versions, `.el9`: can at least build `man`
- `.el8`: nothing (packages for `.el8` don't even contain manpages)

Note that you may not be able to correctly build `html` on Ubuntu 22.04 (currently latest LTS, but in 2 months there will be 24.04 LTS).

**Comment by dimakuv** [comment]
> One solution would be to downgrade docutils to `<0.18`, as it were.

This feels like the easiest solution, why not do that?

**Comment by woju** [comment]
Because it will cause a situation, where you can't simply `apt-get install` all dependencies that will reflect what RTD is doing when building our docs, so you won't really be able to write docs and render them exactly as RTD does. You'll have to mess with virtual environments, which is not something that I would like to suggest to people.

However it's not a regression, we're already in such a situation (hence the bug in the first place).

**Comment by woju** [comment]
#1784 is a draft that fixes the rendering issue and also drops older distros. The downside is, it cannot be iterated on Ubuntu. If we agree on it, we can do #1784, if not, I can do `<` to docutils.

**Comment by mkow** [comment]
This was fixed by #1784, closing.
